src/App.tsx

```diff
@@ -14,78 +14,65 @@ import UserInfo from './pages/UserInfo';
 import ValidateEmail from './pages/ValidateEmail';
 import Game from './pages/Game';
 import SinglePlayer from './pages/SinglePlayer';
-import UserContext from './context/Context';
 import ProtectedRoutes from './components/ProtectedRoutes';
-import { User } from './types/AppTypes';
+import { RootState } from './store';
 
 export const socket = io('http://localhost:5000');
 function App() {
-  const [user, setUser] = useState<User>({
-    username: '',
-    name: '',
-    email: '',
-    loggedIn: true,
-  });
+  const { loggedIn } = useSelector((state: RootState) => state.user);

```

Comment:
Hmm... looks like you have a hooks.ts in your ./store. Which means you could import { useAppSelector } from './store/hooks' on line 4, remove line 18, and change line 22 to something like const { loggedIn } = useAppSelector((state) => state.user). Might as well use tools that appear to be in the project already.